src/strategies/spidStrategy.ts

```diff
@@ -11,6 +11,24 @@ import { SpidUser } from "../types/user";
 import { fetchIdpMetadata, parseIdpMetadata } from "../utils/idpLoader";
 import { log } from "../utils/logger";
 
+export const remapIpdMetadata = (
+  // tslint:disable-next-line: no-any
+  options: { idp: { [key: string]: IDPEntityDescriptor }; sp: any },
+  idpMetadata: ReadonlyArray<IDPEntityDescriptor>,
+  idpIds: { [key: string]: string | undefined }
+) => {
+  idpMetadata.forEach(idp => {
```

Comment:
modifying a parameter is not a good practice, I suggest to have this function return a new `options.idp` object and having the caller merging this new object with the original `options.idp`, i.e.:

```ts
const idpsMetadataOption = remapIpdMetadata(idpMetadata, idpIds);
const options = {
  idp: {
    ...idpsMetadataOption,
    xx_testidp: { ... }
  }
  // ...
}
```

Reply:
fixed in 7d7e64b

Reply:
can you replace the `forEach` with a `reduce`?

Reply:
fixed in 1fb05c8